components/scream/src/physics/p3/atmosphere_microphysics.hpp

```diff
@@ -243,6 +243,13 @@ class P3Microphysics : public AtmosphereProcess
       } // for ipack
       precip_liq_surf_mass(icol) += precip_liq_surf_flux(icol) * PC::RHO_H2O * m_dt;
       precip_ice_surf_mass(icol) += precip_ice_surf_flux(icol) * PC::RHO_H2O * m_dt;
+
+      // Set appropriate boundary fluxes for energy and mass conservation checks.
+      // Any boundary fluxes not included in P3 interface are set to 0.
+      vapor_flux(icol) = 0.0;
```

Comment:
Still related to fields not being present when checks are off: this code should be conditional to conservation checks being on.